data/src/main/java/org/apache/iceberg/data/parquet/GenericParquetWriter.java

```diff
@@ -117,6 +120,13 @@ private GenericParquetWriter() {
     @Override
     public ParquetValueWriter<?> primitive(PrimitiveType primitive) {
       ColumnDescriptor desc = type.getColumnDescription(currentPath());
+      LogicalTypeAnnotation logicalType = primitive.getLogicalTypeAnnotation();
+      if (logicalType != null) {
+        Optional<PrimitiveWriter<?>> writer = logicalType.accept(new LogicalTypeWriterVisitor(desc));
+        if (writer.isPresent()) {
+          return writer.get();
+        }
```

Comment:
When writer.isPresent() == false, we falls back to the previous logic, right?

Reply:
Correct.